Google.Cloud.AspNetCore/Google.Cloud.AspNetCore.DataProtection.Storage/CloudStorageXmlRepository.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Copyright 2019 Google LLC
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     https://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Cloud.Storage.V1;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.DataProtection.Repositories;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Xml;
+using System.Xml.Linq;
+
+using Object = Google.Apis.Storage.v1.Data.Object;
+
+namespace Google.Cloud.AspNetCore.DataProtection.Storage
+{
+    /// <summary>
+    /// Implementation of <see cref="IXmlRepository"/> that stores the protected elements in a single Google Cloud Storage file.
```

Comment:
Protected?  My understanding was that the IXmlRepository class is that it's a general class for storing XML.  The [docs](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.dataprotection.repositories.ixmlrepository?view=aspnetcore-2.2) say simply _The basic interface for storing and retrieving XML elements._ We happen to use the implementation to store encrypted stuff, but I see nothing about the interface that states or implies protection.

Reply:
Well the namespace is Microsoft.AspNetCore.DataProtection.Repositories so I assume it's to do with data protection. Happy to change it if we ever want. (This is an internal comment.)